Configure

```diff
@@ -1662,12 +1662,16 @@ if ($shlib_version_number =~ /(^[0-9]*)\.([0-9\.]*)/)
 	}
 
 my $ecc = $cc;
-$ecc = "clang" if `$cc --version 2>&1` =~ /clang/;
+{
+my $ccv = `$cross_compile_prefix$cc --version 2>&1`;
+$ecc = "clang" if $ccv =~ /clang/;
+$ecc = "gcc" if $ccv =~ /gcc/;
+}
 
 if ($strict_warnings)
 	{
 	my $wopt;
-	die "ERROR --strict-warnings requires gcc or clang" unless ($ecc =~ /gcc$/ or $ecc =~ /clang$/);
+	die "ERROR --strict-warnings requires gcc or clang" unless ($ecc eq "gcc" or $ecc eq "clang");
```

Comment:
Note that on (new) line 1679, there already is an exact string comparison for clang...